### DISCUSSION TITLE
Implementing Interface's method on Union state

### DISCUSSION THREAD

**Discussion by ericodarmawanh** [rrousselGit/freezed#745] [Q&A]
Let's say that I have an interface IError which has a method to implement like this:
![FavJrMfVEAModH3](https://user-images.githubusercontent.com/69150523/186479764-2e2a9643-66dc-4c4e-a20d-b089faba11aa.jpeg)
Now, one of my Union's state implements it. The question is: Is this the right way to implement the method? I doubt my self because now the success state has the method too. It's happened because I create the method inside the Union class.
![FavJvs_UYAApl5p](https://user-images.githubusercontent.com/69150523/186481566-17d439f8-04ad-44e0-9058-6b658115db6b.jpeg)
Thank you for the answer.

**Comment by rrousselGit** [comment]
Hello!

Yes, there's no choice. If the getMessage wasn't part of the IError interface, you could use an extension instead. But because of the interface, you'll have to it as you've shown.
